packages/grid/x-data-grid/src/internals/components/base/GridOverlayWrapper.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/material/utils';
+import { useGridSelector } from '../../hooks/utils/useGridSelector';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { gridDensityHeaderHeightSelector } from '../../hooks/features/density/densitySelector';
+import { GridEvents } from '../../models/events';
+
+export function GridOverlayWrapper(props: React.PropsWithChildren<{}>) {
+  const apiRef = useGridApiContext();
+  const rootProps = useGridRootProps();
+  const headerHeight = useGridSelector(apiRef, gridDensityHeaderHeightSelector);
+
+  const [viewportInnerSize, setViewportInnerSize] = React.useState(
+    () => apiRef.current.getRootDimensions()?.viewportInnerSize ?? null,
+  );
+
+  const handleViewportSizeChange = React.useCallback(() => {
+    setViewportInnerSize(apiRef.current.getRootDimensions()?.viewportInnerSize ?? null);
+  }, [apiRef]);
+
+  useEnhancedEffect(() => {
+    return apiRef.current.subscribeEvent(
+      GridEvents.viewportInnerSizeChange,
+      handleViewportSizeChange,
+    );
+  }, [apiRef, handleViewportSizeChange]);
+
+  let height: React.CSSProperties['height'] = viewportInnerSize?.height ?? 0;
+  if (rootProps.autoHeight && height === 0) {
+    height = 'auto';
+  }
+
+  if (!viewportInnerSize) {
+    return null;
+  }
+
+  return (
+    <div
+      style={{
+        height,
+        width: viewportInnerSize?.width ?? 0,
+        position: 'absolute',
+        top: headerHeight,
+        bottom: 0,
+        left: 0,
```

Comment:
Not needed.

```suggestion
```

Reply:
`bottom` is needed when `height` is `auto`, because otherwise overlay won't take the full height and some tests will fail

Reply:
True, I forgot https://github.com/mui/mui-x/pull/3832#discussion_r811991220. Then, maybe keep `bottom=0` only when `height=auto`, otherwise it's `undefined`. That way we don't have two properties competing.

Reply:
Sure, done